lib/rules/indent.js

```diff
@@ -1094,6 +1094,19 @@ module.exports = {
                     parameterParens.add(closingParen);
                     addElementListIndent(node.params, openingParen, closingParen, options.FunctionExpression.parameters);
                 }
+
+                if (firstToken.type === "Identifier" && firstToken.value === "async") {
+                    const nextToken = sourceCode.getTokenAfter(firstToken);
+
+                    if (astUtils.isOpeningParenToken(nextToken)) {
+                        const closingParen = sourceCode.getTokenBefore(node.body, astUtils.isClosingParenToken);
+
+                        parameterParens.add(nextToken);
+                        parameterParens.add(closingParen);
+                        addElementListIndent(node.params, nextToken, closingParen, options.FunctionExpression.parameters);
+                    }
```

Comment:
This is basically the same code as the code for the first token, the only difference is that it's checking the second token instead of the first token. Could we maybe avoid duplicating these lines? 

Arrow function nodes have `async` boolean property, so maybe we can do something like this:

```js
const maybeOpeningParen = sourceCode.getFirstToken(node, node.async ? 1 : 0);

if (astUtils.isOpeningParenToken(maybeOpeningParen)) {
  // ...
}
```

Reply:
Awesome !

Thanks a lot for sharing this.